src/main.rs

```diff
@@ -0,0 +1,447 @@
+    }
+
+    // Step 11: Give ChannelMonitors to ChainMonitor
+    for item in chain_listener_channel_monitors.drain(..) {
+        let channel_monitor = item.1.0;
+        let funding_outpoint = item.2;
+        chain_monitor.watch_channel(funding_outpoint, channel_monitor).unwrap();
+    }
+
+    // Step 13: Optional: Initialize the NetGraphMsgHandler
+    // XXX persist routing data
+    let genesis = genesis_block(args.network).header.block_hash();
+    let router = Arc::new(NetGraphMsgHandler::new(genesis, None::<Arc<dyn chain::Access>>, logger.clone()));
+
+    // Step 14: Initialize the PeerManager
+    let channel_manager: Arc<ChannelManager> = Arc::new(channel_manager);
+	  let mut ephemeral_bytes = [0; 32];
+	  rand::thread_rng().fill_bytes(&mut ephemeral_bytes);
+    let lightning_msg_handler = MessageHandler { chan_handler: channel_manager.clone(),
+                                                 route_handler: router.clone() };
+    let peer_manager: Arc<PeerManager> = Arc::new(PeerManager::new(lightning_msg_handler,
+                                                        keys_manager.get_node_secret(),
+                                                        &ephemeral_bytes, logger.clone()));
+
+    // ## Running LDK
+    // Step 16: Initialize Peer Connection Handling
+
+    // We poll for events in handle_ldk_events(..) rather than waiting for them over the
+    // mpsc::channel, so we can leave the event receiver as unused.
+    let (event_ntfn_sender, mut _event_ntfn_receiver) = mpsc::channel(2);
+    let peer_manager_connection_handler = peer_manager.clone();
+    let event_notifier = event_ntfn_sender.clone();
+    let listening_port = args.ldk_peer_listening_port;
+    runtime.spawn(async move {
+	      let listener = std::net::TcpListener::bind(format!("0.0.0.0:{}", listening_port)).unwrap();
+        loop {
+            let tcp_stream = listener.accept().unwrap().0;
+            lightning_net_tokio::setup_inbound(peer_manager_connection_handler.clone(),
+                                               event_notifier.clone(), tcp_stream).await;
+        }
+    });
+
+    // Step 17: Connect and Disconnect Blocks
+    if chain_tip.is_none() {
+        chain_tip = Some(runtime.block_on(init::validate_best_block_header(&mut bitcoind_rpc_client)).unwrap());
+    }
+    let channel_manager_listener = channel_manager.clone();
+    let chain_monitor_listener = chain_monitor.clone();
+    let network = args.network;
+    runtime.spawn(async move {
+        let chain_poller = poll::ChainPoller::new(&mut bitcoind_rpc_client, network);
+        let chain_listener = (chain_monitor_listener, channel_manager_listener);
+        let mut spv_client = SpvClient::new(chain_tip.unwrap(), chain_poller, &mut cache,
+                                            &chain_listener);
+        loop {
+            spv_client.poll_best_tip().await.unwrap();
+            thread::sleep(Duration::new(1, 0));
```

Comment:
We shouldn't sleep/loop inside a tokio task, instead we should use tokio's timers to fire an event and poll every time we get it.

Reply:
I talked to the tokio devs about this, and it's fine as long as we `await` regularly within the loop, you just can't block for a long time ("block" = refers to how much time is spent between calls to `awati`). Here's a link to the convo: https://discord.com/channels/500028886025895936/500336333500448798/822583155361185852

Reply:
Right, I assume tokio::sleep is fine, thread::sleep not so much.